### DISCUSSION TITLE
Upgraded Soketi to uWebSocket 20.43.0 with Major improvements.

### DISCUSSION THREAD

**Discussion by ehsanR91** [soketi/pws#1122] [Show and tell]
Hello Everyone,

I have seen that this project was not getting the attention it deserved and the author didn't upgraded the core engine. I have made allot of changes to Soketi and i am planning to implement allot of more changes. As of now, i have made the following changes:

Upgrade core uWebSocket to 20.43.0 (Latest)
Support for Node 22,23 
Subscription Count for individual channels (now whenever a user no matter of which type connects to a channel, you can get total number of counts for that channel in webhook)
Support for Max Data.
Implemented deflate compression (55% savings on JSON data types).

Work in progress:

Implementing customized UI written in Go which can be packaged alongside.
Improve overall code quality.
Add unit testing.
Ability to hot-reload with separate config.json reloads.

I will be pushing the changed version to my github in a day or two if i feel like people need it.

**Comment by ehsanR91** [comment]
I have released the first initial release with all major upgrades, this includes all dependies upgraded to latest as below 

pusher-js -> 8.4.0
uWebSockersJs -> 20.43.0
uuid -> 9.0
sqs-consumer -> 9.1.0
axios -> 1.6.7
bullmq -> 5.4.2 (major upgrade)
knex -> 3.1.0

and almost all the dependcies have been now upgraded to their latest.

I have released the first official package on my github, you can use distroless docker file and get it all going in a quick go.

**Comment by ramiel** [comment]
Hello @ehsanR91,
since soketi project is dead, I think the community should bring its effort and sustain your variant. It may become a first citizen project, maybe even with a new name, a website and so on. Let's see if anybody else would like to achieve that.

**Reply by ehsanR91** [reply]
I would be more than happy if community wants to pitch-in and bring life back to this project. We can achieve great things with it, i am ready to even invest, time, effort and some finance into it aswell. Regarding name, if there's any suggestion, that would be great.

**Reply by Faolain** [reply]
@rennokki would you be interested in having @ehsanR91 as a collaborator or integrating his changes? Or would you prefer a fork? The community would prefer your knowledge and expertise.
